operations/network/management_network/canu/custom_config.md

```diff
@@ -0,0 +1,146 @@
+    snmp-server user testuser v3 capability admin
+    snmp-server user testuser v3 enable
+    snmp-server user testuser v3 enable sets
+    snmp-server user testuser v3 encrypted auth md5 xxxxxxx priv des xxxxxxx
+    snmp-server user testuser v3 require-privacy
+sw-spine-002:  |
+    interface ethernet 1/16 speed 10G force
+    interface ethernet 1/16 description "sw-spine01-1/16"
+    interface ethernet 1/16 no switchport force
+    interface ethernet 1/16 ip address 10.102.255.34/30 primary
+    interface ethernet 1/16 dcb priority-flow-control mode on force
+    ip route vrf default 0.0.0.0/0 10.102.255.33
+    snmp-server user testuser v3 capability admin
+    snmp-server user testuser v3 enable
+    snmp-server user testuser v3 enable sets
+    snmp-server user testuser v3 encrypted auth md5 xxxxxxx priv des xxxxxxx
+    snmp-server user testuser v3 require-privacy
+sw-leaf-bmc-001:  |
+    interface ethernet1/1/12
+      description sw-leaf-bmc-001:12==>cn003:2
+      no shutdown
+      switchport access vlan 4
+      mtu 9216
+      flowcontrol receive off
+      flowcontrol transmit off
+      spanning-tree bpduguard enable
+      spanning-tree port type edge
+    interface vlan7
+        description CMN
+        no shutdown
+        ip vrf forwarding Customer
+        mtu 9216
+        ip address 10.102.4.100/25
+        ip access-group cmn-can in
+        ip access-group cmn-can out
+        ip ospf 2 area 0.0.0.0
+    snmp-server group cray-reds-group 3 noauth read cray-reds-view
+    snmp-server user testuser cray-reds-group 3 auth md5 xxxxxxxx priv des xxxxxxx
+    snmp-server view cray-reds-view 1.3.6.1.2 included
+```
+
+To generate switch configuration with custom config injection.
+
+```text
+canu generate switch config --csm 1.2 -a full --shcd FILENAME.xlsx --tabs INTER_SWITCH_LINKS,NON_COMPUTE_NODES,HARDWARE_MANAGEMENT,COMPUTE_NODES --corners J14,T44,J14,T48,J14,T24,J14,T23 --sls-file SLS_FILE --name sw-spine-001 --custom-config CUSTOM_CONFIG_FILE.yaml
+```
+
+## Generate Network Config
+
+CANU can also generate switch config for all the switches on a network.
+
+In order to generate network config, a valid SHCD or CCJ must be passed in and system variables must be read in from either CSI output or the SLS API. The instructions are exactly the same as the above `generate Switch Config](#generate-switch-config)` except there will not be a hostname and a folder must be specified for config output using the `--folder FOLDERNAME` flag.
+
+To generate switch config from a CCJ paddle run:
+
+```text
+canu generate network config --csm 1.2 --ccj paddle.json --sls-file SLS_FILE --folder FOLDERNAME
+```
+
+To generate switch config from SHCD run:
+
+```text
+canu generate network config --csm 1.2 -a full --shcd FILENAME.xlsx --tabs INTER_SWITCH_LINKS,NON_COMPUTE_NODES,HARDWARE_MANAGEMENT,COMPUTE_NODES --corners J14,T44,J14,T48,J14,T24,J14,T23 --sls-file SLS_FILE --folder switch_config
+
```

Comment:
nit: Typically a snippet like this is encouraged to be broken up into a command and example output snippet. If you feel it's more appropriate as-is then disregard.